src/hydra-eval-jobs/hydra-eval-jobs.cc

```diff
@@ -504,6 +504,16 @@ int main(int argc, char * * argv)
 
             job.erase("namedConstituents");
 
+            /* Register the derivation as a GC root.  !!! This
+                registers roots for jobs that we may have already
+                done. */
+            auto localStore = store.dynamic_pointer_cast<LocalFSStore>();
+            if (gcRootsDir != "" && localStore) {
+                Path root = gcRootsDir + "/" + std::string(baseNameOf((std::string) job["drvPath"]));
```

Comment:
I can't find where the `Path` type comes from, but I'd ask if this is the proper way to concatenate paths for this type, as there may be a default path separator character we'd want to use? 

if this path came from `std::filesystem::path` we'd be able to use the `/ operator` overload to concatenate the path elements and I'd assume that this path type would have something similar, or some expectation of what "proper data" would look like.

Reply:
Good question. It looks like this is a `nix::Path`, but I couldn't find where that was defined.

Reply:
It looks like `gcRootsDir` could be made a std::filesystem::path, though? I don't know the right way to do that.

Reply:
(FWIW, `Path` is defined here: https://github.com/NixOS/nix/blob/5b809f9e0e0fe84304c2ae0f5f7b2d4db02565ad/src/libutil/types.hh#L25)